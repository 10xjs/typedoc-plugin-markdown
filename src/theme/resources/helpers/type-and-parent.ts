import * as Handlebars from 'handlebars';
import { SignatureReflection } from 'typedoc';
import { ArrayType, ReferenceType } from 'typedoc/dist/lib/models/types';

export function typeAndParent(this: ArrayType | ReferenceType) {
  if (this instanceof ReferenceType && this.reflection) {
    const md = [];
    if (this.reflection instanceof SignatureReflection) {
      if (this.reflection.parent.parent.url) {
        md.push(
          `[${this.reflection.parent.parent.name}](${Handlebars.helpers.relativeURL.call(
            this,
            this.reflection.parent.parent.url,
          )})`,
        );
      } else {
        md.push(this.reflection.parent.parent.name);
      }
    } else {
      if (this.reflection.parent.url) {
        md.push(
          `[${this.reflection.parent.name}](${Handlebars.helpers.relativeURL.call(this, this.reflection.parent.url)})`,
        );
      } else {
        md.push(this.reflection.parent.name);
      }
      if (this.reflection.url) {
        md.push(`[${this.reflection.name}](${Handlebars.helpers.relativeURL.call(this, this.reflection.url)})`);
      } else {
        md.push(this.reflection.name);
      }
    }
    return md.join('.');
  }
  return 'void';
}
