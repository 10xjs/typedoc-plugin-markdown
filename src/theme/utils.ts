import * as fs from 'fs';
import * as Handlebars from 'handlebars';
import * as path from 'path';
import { DeclarationReflection, Reflection } from 'typedoc/dist/lib/models/reflections/index';

export function compileTemplate(template: string, data: {}) {
  const source = templateToString(template);
  const md: HandlebarsTemplateDelegate = Handlebars.compile(source);
  return new Handlebars.SafeString(md(data));
}

export function safeString(data: string) {
  return new Handlebars.SafeString(data);
}

export function getAnchorRef(ref: string) {
    return ref.replace(/_|\/|\.| /g, '-').replace(/"/g, '').replace(/ /g, '-').toLowerCase();
  }

function templateToString(template: string) {
  return fs.readFileSync(path.join(__dirname, `/${template}`)).toString();
}
