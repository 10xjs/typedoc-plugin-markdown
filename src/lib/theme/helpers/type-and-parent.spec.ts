import { Fixture, getFixture, getReflectionByName } from '../../../test/test.utils';
import { typeAndParent } from './type-and-parent';

describe(`typeAndParent helper`, () => {
  test(`should compile`, () => {
    const data = getReflectionByName(getFixture(Fixture.Classes), 'BaseClass');
    const result = typeAndParent.call(data.children[3].implementationOf);
    expect(result).toMatchSnapshot();
  });
});
