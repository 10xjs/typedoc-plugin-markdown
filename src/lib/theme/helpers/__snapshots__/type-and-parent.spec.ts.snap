// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typeAndParent helper should compile 1`] = `"void"`;
